OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -56,5 +58,13 @@ protected override void OnInside(Actor self)
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
 					notification, self.Owner.Faction.InternalName);
 		}
+
+		public override Activity Tick(Actor self)
+		{
+		if (self.Trait<Infiltrates>().IsTraitDisabled)
```

Comment:
I don't think it is valid to do a trait lookup here.

Which `Infiltrates` will this return?

Can/should this be passed in the ctor?

ping @pchote